lib/pbench/test/unit/server/test_server_configuration.py

```diff
@@ -0,0 +1,220 @@
+        assert response.json == {
+            "message": "Unrecognized keyword ['fookey'] given for parameter key; allowed keywords are ['dataset-lifetime', 'server-banner', 'server-state']"
+        }
+
+    def test_put_bad_keys(self, query_put):
+        response = query_put(
+            key=None,
+            json={"config": {"fookey": "bar"}},
+            expected_status=HTTPStatus.BAD_REQUEST,
+        )
+        assert response.json == {
+            "message": "Unrecognized JSON key ['fookey'] given for parameter config; allowed keywords are ['dataset-lifetime', 'server-banner', 'server-state']"
+        }
+
+    @pytest.mark.parametrize(
+        "key,value",
+        [
+            ("dataset-lifetime", "14 years"),
+            ("server-banner", {"banner": None}),
+            ("server-state", "running"),
+            ("server-state", {"status": "disabled"}),
+        ],
+    )
+    def test_bad_value(self, query_put, key, value):
+        response = query_put(
+            key=key, expected_status=HTTPStatus.BAD_REQUEST, json={"value": value}
+        )
+        assert (
+            f"Unsupported value for configuration key '{key}'"
+            in response.json["message"]
+        )
+
+    def test_put_param(self, query_put):
+        response = query_put(key="dataset-lifetime", query_string={"value": "2"})
+        assert response.json == {"dataset-lifetime": "2"}
+
+    def test_put_value(self, query_put):
+        response = query_put(key="dataset-lifetime", json={"value": "2"})
+        assert response.json == {"dataset-lifetime": "2"}
+
+    def test_put_value_unauth(self, query_put):
+        response = query_put(
+            key="dataset-lifetime",
+            token="",
+            json={"value": "4 days"},
+            expected_status=HTTPStatus.UNAUTHORIZED,
+        )
+        assert response.json == {
+            "message": "Unauthenticated client is not authorized to UPDATE a server administrative resource"
+        }
+
+    def test_put_value_user(self, query_put, pbench_token):
+        response = query_put(
+            key="dataset-lifetime",
+            token=pbench_token,
+            json={"value": "4 days"},
+            expected_status=HTTPStatus.FORBIDDEN,
+        )
+        assert response.json == {
+            "message": "User drb is not authorized to UPDATE a server administrative resource"
+        }
+
+    def test_put_config(self, query_get, query_put):
+        response = query_put(
+            key=None,
+            expected_status=HTTPStatus.OK,
+            json={
+                "config": {
+                    "dataset-lifetime": "2",
+                    "server-state": {"status": "enabled"},
+                }
+            },
+        )
+        assert response.json == {
+            "dataset-lifetime": "2",
+            "server-state": {"status": "enabled"},
+            "server-banner": None,
+        }
+        response = query_get({}, HTTPStatus.OK)
```

Comment:
Why are we using `{}` for the first parameter instead of `None`?  (I think the latter would be better for a couple of reasons.)

Reply:
That's .... a very good question. 😕